lib/Mailer/Mailjet.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Stampie\Mailer;
+
+use Stampie\Adapter\ResponseInterface;
+use Stampie\Attachment;
+use Stampie\Exception\ApiException;
+use Stampie\Exception\HttpException;
+use Stampie\IdentityInterface;
+use Stampie\Mailer;
+use Stampie\Message\AttachmentsAwareInterface;
+use Stampie\Message\TaggableInterface;
+use Stampie\MessageInterface;
+use Stampie\Util\AttachmentUtils;
+
+/**
+ * @author Jérôme Parmentier <jerome@prmntr.me>
+ */
+class Mailjet extends Mailer
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getEndpoint()
+    {
+        return 'https://api.mailjet.com/v3.1/send';
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function setServerToken($serverToken)
+    {
+        if (false === strpos($serverToken, ':')) {
+            throw new \InvalidArgumentException('Mailjet uses a "publicApiKey:privateApiKey" based ServerToken');
+        }
+
+        parent::setServerToken($serverToken);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getHeaders()
+    {
+        return [
+            'Accept' => 'application/json',
+            'Authorization' => sprintf('Basic %s', base64_encode($this->getServerToken())),
+            'Content-Type' => 'application/json',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function format(MessageInterface $message)
+    {
+        $headers = [];
+        foreach ($message->getHeaders() as $name => $value) {
+            $headers[$name] = $value;
```

Comment:
this is useless. You are just doing `$headers = $message->getHeaders()` but forcing PHP to duplicate the array in memory early instead of using copy-on-write. So just use the better version of it

Reply:
I can't find a decent justification for this. :smile: